lib/Config.js

```diff
@@ -467,6 +373,15 @@ function locationConstraintAssert(locationConstraints) {
             assert(typeof details.credentials.secretKey === 'string',
                 'bad config: credentials must include secretKey as string');
         }
+
+        if (locationConstraints[l].type === 'tlp') {
+            // should have no detail
+            assert(Object.keys(locationConstraints[l].details).length === 0);
+
+            // should be a cold location
+            assert(locationConstraints[l].isCold === true);
+        }
```

Comment:
Why do we stop validating the locations here? is it because we rely on the validation at the location creation time? If yes, can we put a comment?

Reply:
what do you mean "stop" validating locations?
* we keep the whole validation, there is no early exit
* we do not store any details anymore (as they are useless for cloudserver, and contain secrets), so the validation of these details is now removed, and we now check that `details` is indeed empty
* the block was just moved early to simplify testing no details are provided (since default fields are added later in the process)